### PR TITLE
MODQM-405 Add allowed values for fields in marc specification.

### PR DESCRIPTION
**Pull request by pawelmol**
### Purpose

Change marc specification for 008 and MARC_BIBLIOGRAPHIC, added "allowedValues" property in some items from "items" array.
https://folio-org.atlassian.net/browse/MODQM-405

Also change "isArray" to false for "Proj" item for "maps" document type.
https://folio-org.atlassian.net/browse/MODQM-406

### Approach

- Update marc specification stored in db to new one with "allowedValues" property set
- Bump up marc-specification interface version
- Update integration tests

### CODE REVIEW

**Comment by psmagin** [comment]
Hi @pawelmol, please follow the PR template https://github.com/folio-org/mod-quick-marc/blob/master/PULL_REQUEST_TEMPLATE.md
Also please follow https://folio-org.atlassian.net/wiki/spaces/FOLIJET/pages/1400654/Conventional+Commit+Initiative when you will merge this changes to master.

File: pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
   <groupId>org.folio</groupId>
   <artifactId>mod-quick-marc</artifactId>
-  <version>5.1.0-SNAPSHOT</version>
+  <version>5.1.1-SNAPSHOT</version>
```
Comment (psmagin) [inline pom.xml:15]: You don't need to change version of the module. 
You should bump specification interface version.

File: src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  <changeSet id="MODQM-405@@update-marc-specification-bibliographic" author="phabas">
+    <preConditions onFail="MARK_RAN">
+      <sqlCheck expectedResult="1">
+        SELECT COUNT(id)
+        FROM marc_specification
+        WHERE field_tag = '008'
+          and record_type = 'MARC_BIBLIOGRAPHIC';
+      </sqlCheck>
+    </preConditions>
+
```
Comment (psmagin) [inline src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml:16]: Please add a comment section

File: src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      <column name="marc_spec" valueClobFile="field008_bibliographic.json"/>
+    </insert>
+  </changeSet>
+
+  <!--Liquibase have some issues with loading valueClobFile into jsonb datatype so need to change type after loading json-->
+  <changeSet id="MODQM-405@@alter-jsonb-marc-specification-table" author="phabas">
+    <sql dbms="postgresql">
+      ALTER TABLE marc_specification ALTER COLUMN marc_spec TYPE JSONB USING marc_spec::JSONB;
+    </sql>
+  </changeSet>
```
Comment (psmagin) [inline src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml:37]: Why do you have 3 different queries in one changest and 4th one in separate changeset? What is a purpose for this?

**Review by psmagin — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (pawelmol) [reply pom.xml]: ok, done

File: src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml (hunk 2 above)
Comment (pawelmol) [reply src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml:16]: I'm not sure what do you mean, I added a xml style comment with some summary.

File: src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml (hunk 3 above)
Comment (pawelmol) [reply src/main/resources/db/changelog/changes/v5.1.1/update-marc-specification-bibliographic.xml]: Right it was not necessary/or even bug, this should be run as one transaction.

File: src/main/resources/db/changelog/changelog-5.1.1.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+
+    <include file="changes/v5.1.1/update-marc-specification-bibliographic.xml" relativeToChangelogFile="true"/>
```
Comment (psmagin) [inline src/main/resources/db/changelog/changelog-5.1.1.xml:7]: It shouldn't be now 5.1.1

**Review by psmagin — CHANGES_REQUESTED** [review]
Did you see my comment about PR template?

File: src/main/resources/db/changelog/changelog-5.1.1.xml (hunk 4 above)
Comment (pawelmol) [reply src/main/resources/db/changelog/changelog-5.1.1.xml]: ok, switched to 5.1.0

**Review by psmagin — APPROVED** [review]
Please update NEWS file before merging

**Review by TsaghikKhachatryan — APPROVED** [review]

**Pull request merged**